packages/react-storage/src/components/StorageBrowser/views/LocationActionView/hooks/useDeleteActionView.tsx

```diff
@@ -0,0 +1,224 @@
+              content: {
+                icon: getActionIconVariant(item.status),
+                text: getFilenameWithoutPrefix(item.key),
+              },
+            };
+          }
+          case 'folder': {
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: path,
+              },
+            };
+          }
+          case 'type': {
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: getFileTypeDisplayValue(item.key),
+              },
+            };
+          }
+          case 'size':
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: humanFileSize(
+                  parseInt((item as Task & { size: string }).size),
+                  true
+                ),
+              },
+            };
+          case 'status':
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: STATUS_DISPLAY_VALUES[item.status],
+              },
+            };
+          case 'action':
+            // don't allow removing a single task
+            if (taskCounts.TOTAL > 1) {
+              return hasStarted
+                ? {
+                    key,
+                    type: 'button',
+                    content: {
+                      icon: 'cancel',
+                      ariaLabel: `Cancel item: ${item.key}`,
+                      onClick: () => item.cancel?.(),
+                      disabled: item.status !== 'QUEUED',
+                    },
+                  }
+                : {
+                    key,
+                    type: 'button',
+                    content: {
+                      icon: 'cancel',
+                      ariaLabel: `Remove item: ${item.key}`,
+                      onClick: () => item.remove(),
+                    },
+                  };
+            } else {
+              return {
+                key,
+                type: 'text',
+                content: {
+                  text: '',
+                },
+              };
+            }
+          default:
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: '',
+              },
+            };
+        }
+      }),
+    };
+    return row;
+  });
+
+  const tableData: DataTableProps = {
+    headers: LOCATION_ACTION_VIEW_HEADERS,
+    rows,
+  };
+  return tableData;
+};
+
+export const useDeleteActionView = ({
+  onClose: _onClose,
+}: {
+  onClose?: () => void;
+}): UseDeleteActionView => {
+  const [
+    {
+      history,
+      locationItems: { fileDataItems: selected },
+    },
+    dispatchStoreAction,
+  ] = useStore();
+  const { current } = history;
+  const path = current?.prefix;
+  const getInput = useGetActionInput();
+
+  const [tasks, handleProcess] = useProcessTasks(
+    // @ts-expect-error
```

Comment:
Leaving this error here per conversation with @calebpollman because there is ongoing work to refactor the data type, which is causing the error